src/table.js

```diff
@@ -1381,6 +1382,88 @@ Table.prototype.truncate = function(gaxOptions, callback) {
   );
 };
 
+/**
+ * Generates Consistency-Token and check consistency for generated token
+ * In-case consistency check returns false, retrial is done in interval
+ * of 5 seconds till 10 minutes, after that it returns false.
+ *
+ * @param {function(?error, ?boolean)} callback The callback function.
+ * @param {?Error} callback.err An error returned while making this request.
+ * @param {?Boolean} callback.resp Boolean value.
+ */
+Table.prototype.waitForReplication = function(callback) {
+  const reqOpts = {
+    name: this.id,
+  };
+
+  this.bigtable.request(
+    {
+      client: 'BigtableTableAdminClient',
+      method: 'generateConsistencyToken',
+      reqOpts: reqOpts,
+    },
+    (err, resp) => {
+      if (err) {
+        callback(err);
+        return;
+      }
+
+      const timeoutAfterTenMinutes = setTimeout(() => {
+        callback(null, false);
+      }, 10 * 60 * 1000);
+
+      // method to launch token consistency check
+      const launchCheck = () => {
+        const token = resp[0].consistencyToken;
+        this.checkConsistency(token, retryIfNecessary);
+      };
+
+      // method checks if retrial is required & init retrial with 5 sec delay
+      const retryIfNecessary = (err, res) => {
+        if (err || res === true) {
+          clearTimeout(timeoutAfterTenMinutes);
+          callback(err, res);
+          return;
+        }
+
+        setTimeout(launchCheck, 5000);
+      };
+
+      launchCheck();
+    }
+  );
+};
+
+/**
+ * Checks consistency for given ConsistencyToken
+ *
+ * @param {function(?error, ?boolean)} callback The callback function.
+ * @param {?Error} callback.err An error returned while making this request.
+ * @param {?Boolean} callback.resp Boolean value.
+ */
+Table.prototype.checkConsistency = function(token, callback) {
```

Comment:
If this is going to be a public method, then there needs to be another public method to retrieve a token via `generateConsistencyToken`

Reply:
Ok, 'll do this

Reply:
Done